src/app/@ansyn/ansyn/modules/menu-items/tools/export-maps-popup/export-maps-popup.component.ts

```diff
@@ -1,99 +1,304 @@
-			if (isExport) {
-				saveAs(blob, 'map.jpg');
-				this.logger.info(LOGS.success);
-			}
-			else {
-				this.logger.info(LOGS.canceled);
-			}
-			this.closeModal()
-		})
+	onActiveMapChange$ = this.store$.pipe(
+		select(selectActiveMapId),
+		tap( (activeMapId) => this.pdfExportMapId = activeMapId)
 	);
-
 	@AutoSubscription
-	isMinimalView$ = this.store$.select(selectIsMinimalistViewMode).pipe(
+	basicExport$ = this.store$.pipe(
+		select(selectIsMinimalistViewMode),
 		filter(Boolean),
-		tap( this.exportMapsToPng.bind(this))
+		debounceTime(100),
+		tap(this.basicExportMap.bind(this))
 	);
 
+	_pageSize = DEFAULT_PAGE_SIZE;
+
+	get pageSize() {
+		if (this.isPDF()) {
+			return this._pageSize;
+		}
+		return 'Screenshot';
+	}
+
+	set pageSize(value) {
+		this._pageSize = value
+	}
+
+	_pageSizes = { a0: [1189, 841], a1: [841, 594], a2: [594, 420], a3: [420, 297], a4: [297, 210], a5: [210, 148] };
+
+	get pageSizes() {
+		if (this.isPDF()) {
+			return Object.keys(this._pageSizes);
+		}
+		return ['Screenshot']
+	}
+
+	_qualities = { low: 72, normal: 150, high: 300 };
+
+	get qualities() {
+		return Object.keys(this._qualities);
+	}
+
 	get config() {
 		return this.toolsConfigData.exportMap;
 	}
 
-	constructor(
-		public store$: Store<any>,
-		public credentialsService: CredentialsService,
-		public dialogRef: MatDialogRef<ExportMapsPopupComponent>,
-		protected logger: LoggerService,
-		@Inject(DOCUMENT) protected document: any,
-		@Inject(toolsConfig) public toolsConfigData: IToolsConfig) {
+	constructor(protected store$: Store<any>,
+				protected logger: LoggerService,
+				protected dialogRef: MatDialogRef<ExportMapsPopupComponent>,
+				protected imageryCommunicatorService: ImageryCommunicatorService,
+				protected translateService: TranslateService,
+				@Inject(DOCUMENT) protected document: any,
+				@Inject(toolsConfig) public toolsConfigData: IToolsConfig) {
 		this.logger.info(LOGS.request);
-		this.store$.dispatch(new SetMinimalistViewModeAction(true));
+		this.translateService.get('direction', '').subscribe( (direction) => this.direction = direction === 'rtl' ? 'rtl' : 'ltr')
 	}
 
-	exportMapsToPng() {
-			toBlob(document.querySelector(this.config.target), {
-				filter: (element) => {
-					if (element.tagName === 'CANVAS') {
-						return element.width > 0;
-					}
-					return !(element.classList && this.config.excludeClasses.some( excludeClass => element.classList.contains(excludeClass)));
+	ngOnInit(): void {
+	}
+
+	ngOnDestroy(): void {
+	}
+
+	advancedExportMaps(exportMetadata: IExportMapMetadata) {
+		const mapToBeExport = this.imageryCommunicatorService.provide(this.pdfExportMapId);
+
+		if (!mapToBeExport) {
+			throw new Error('No Such Map Error');
+		}
+
+
+		mapToBeExport.exportMap(exportMetadata).pipe(
+			mergeMap( (exportMapData) => {
+				if (exportMetadata.extra.north) {
+					return new Observable( obs => {
+						const rotation = mapToBeExport.getRotation();
```

Comment:
can you elaborate on the function?

Reply:
on the advance export we use the map object.
for export in better resolution,
so the compass was not there as it not part of the map.
so if the user select to export it i add it manually.